reactor-core/src/main/java/reactor/core/publisher/FluxMergeSequential.java

```diff
@@ -273,8 +273,11 @@ void drainAndCancel() {
 		}
 
 		void cancelAll() {
-			MergeSequentialInner<R> inner;
+			if (this.current != null) {
```

Comment:
Use assign in case of nullification by another path (there are several `inner= null`)

Reply:
good catch, did that initially reusing `inner` but that was less readable, then I forgot about keeping a local variable.